tests/test_modeling_common.py

```diff
@@ -4763,6 +4763,52 @@ def test_torch_compile(self):
         for i in range(n_iter):
             _ = model.generate(**input_ids, do_sample=False)
 
+    @slow
+    @require_torch_gpu
+    def test_torch_compile_for_training(self):
+        if version.parse(torch.__version__) < version.parse("2.3"):
+            self.skipTest(reason="This test requires torch >= 2.3 to run.")
+
+        if not hasattr(self, "_torch_compile_train_cls"):
+            self.skipTest(f"{self.__class__.__name__} doesn't have the attribute `_torch_compile_train_cls`.")
+
+        config, _ = self.model_tester.prepare_config_and_inputs_for_common()
+        cls = self._torch_compile_train_cls
+        batch_size = 1
+        seq_len = 10
+
+        model = cls(config).to(torch_device)
+
+        inputs = {
+            "input_ids": torch.randint(
+                low=1, high=model.config.vocab_size, size=(batch_size, seq_len), device=torch_device
+            ),
+            "attention_mask": torch.ones((batch_size, seq_len), dtype=torch.int64, device=torch_device),
```

Comment:
attn mask being full of one is more prone to skipping some branches, would try with ones and zeros as welll!